roles/default/kiali-deploy/templates/kubernetes/deployment.yaml

```diff
@@ -102,20 +102,18 @@ spec:
           value: "{{ kiali_vars.deployment.logger.sampler_rate }}"
         - name: LOG_TIME_FIELD_FORMAT
           value: "{{ kiali_vars.deployment.logger.time_field_format }}"
-{% for env in kiali_deployment_environment_variables %}
-        - name: {{ env }}
-          valueFrom:
-            secretKeyRef:
-              name: {{ kiali_deployment_environment_variables[env].secret_name }}
-              key: {{ kiali_deployment_environment_variables[env].secret_key }}
-{% endfor %}
         volumeMounts:
         - name: kiali-configuration
           mountPath: "/kiali-configuration"
         - name: kiali-secret
           mountPath: "/kiali-secret"
         - name: kiali-cabundle
           mountPath: "/kiali-cabundle"
+{% for sec in kiali_deployment_secret_volumes %}
+        - name: {{ sec }}
+          mountPath: "/kiali-override-secrets/{{ sec }}"
+          readOnly: true
+{% endfor %}
```

Comment:
Do these need to be separate volumes or can they be combined?

Reply:
Because these are separate secrets, I believe they need to be mounted separately. While I did not actually try it, it looks like the volumes need to be in their own locations. For example, I don't think we can put everything in mountPath "/kiali-override-secrets" because the keys in multiple secrets could conflict and thus cause a problem since the filenames would be the same (or we'd have to somehow figure out a way to create non-conflicting names). For example, if you had two secrets and both have a key "password" in them, if they are both mounted in the same /kiali-override-secrets path, we have no easy way to avoid that conflict. The way this is now, it is easy to configure without worry about conflicts and it ensures each secret is separately mounted in their own location.

Reply:
for the record, I just tried it to see what would happen if I had multiple secrets mounted on the same mountPath (I set all the mountPath settings to "/kiali-override-secrets") but make sure I set all the path values to a unique name (to avoid name conflicts) ... it appears the last one wins. I was expecting to see 7 files, but I only see 1.

Reply:
looks like there is this "projected" feature that might allow to consolidate this all - but I don't see a need to investigate this any further. This is a hidden implementation detail that should not concern the user and it is fine having subdirectories for each secret. Not much benefit here by consolidating this further.